src/classes/BDI_ContactService.cls

```diff
@@ -0,0 +1,1095 @@
+            'Id', 'Firstname', 'Lastname', 'AccountId', 'Account.npe01__SYSTEM_AccountType__c',
+            UTIL_Namespace.StrTokenNSPrefix('Primary_Affiliation__c'),
+            'Email', 'npe01__AlternateEmail__c', 'npe01__HomeEmail__c', 'npe01__WorkEmail__c',
+            'Phone', 'HomePhone', 'MobilePhone', 'OtherPhone', 'npo02__Formula_HouseholdPhone__c', 'npe01__WorkPhone__c' };
+        setContactFields.addAll(mapDIFieldToC1Field.values());
+        setContactFields.addAll(mapDIFieldToC2Field.values());
+        return 'SELECT ' + string.join(new List<String>(setContactFields), ',') + ' FROM Contact ';
+    }
+
+    /*******************************************************************************************************
+    * @description returns the Contact's full name for use in querying
+    * @param di the Data Import record
+    * @param iCon 1 or 2 to specify which contact, eg Contact1 or Contact2
+    * @return String the Contact's full name or null
+    */
+    private String getContactNameFromDi(DataImport__c di, Integer iCon) {
+        String strCx = UTIL_Namespace.StrTokenNSPrefix('Contact' + string.valueOf(iCon));
+        String firstname = (String)di.get(strCx + '_Firstname__c');
+        String lastname = (String)di.get(strCx + '_Lastname__c');
+
+        // note that we don't try to support middlename and suffix due to known
+        // salesforce won't fix issue: https://success.salesforce.com/issues_view?id=a1p30000000eQRxAAM
+        // String middlename = getMiddleName(di, strCx);
+        // String suffix = getNameSuffix(di, strCx);
+
+        String fullname = '';
+        if (firstname != null) {
+            fullname += firstname + ' ';
+        }
+        if (lastname != null) {
+            fullname += lastname;
+        }
+        return (fullname);
+    }
+
+    /*******************************************************************************************************
+    * @description adds a non-null string to the provided set
+    * @param str the string value
+    * @param setStr the set to add to
+    * @return void
+    */
+    private void addStringToSet(String str, Set<String> setStr) {
+        if (str != null) {
+            setStr.add(str);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description adds a non-null Id to the provided set
+    * @param Id the Id value
+    * @param setId the set to add to
+    * @return void
+    */
+    private void addIdToSet(Id id, Set<Id> setId) {
+        if (id != null) {
+            setId.add(id);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description adds a non-null Double to the provided set
+    * @param dbl the Double value
+    * @param setDbl the set to add to
+    * @return void
+    */
+    private void addDoubleToSet(Double dbl, Set<Double> setDbl) {
```

Comment:
Any reason not to combine these three methods into ```void addObjectToSet(Object obj, Set<Object> setObj)```?

Reply:
I had tried that, but it complains about signatures not matching.  So I went with the explicit types, rather than casting.  This was important since a few callers are getting their data from a obj.get() call that needs to be cast to the correct underlying type.